src/lightning/fabric/loggers/wandb.py

```diff
@@ -0,0 +1,701 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from argparse import Namespace
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, List, Literal, Mapping, Optional, Union
+
+import torch.nn as nn
+from lightning_utilities.core.imports import RequirementCache
+from torch import Tensor
+from typing_extensions import override
+
+from lightning.fabric.loggers.logger import Logger, rank_zero_experiment
+from lightning.fabric.utilities.exceptions import MisconfigurationException
+from lightning.fabric.utilities.logger import _add_prefix, _convert_params, _sanitize_callable_params
+from lightning.fabric.utilities.rank_zero import rank_zero_only, rank_zero_warn
+from lightning.fabric.utilities.types import _PATH
+
+if TYPE_CHECKING:
+    from wandb import Artifact
+    from wandb.sdk.lib import RunDisabled
+    from wandb.wandb_run import Run
+
+    from lightning.pytorch.callbacks.model_checkpoint import (
+        ModelCheckpoint,  # TODO: Replace with new Fabric Checkpoints system
+    )
+
+_WANDB_AVAILABLE = RequirementCache("wandb>=0.15.0")
+
+
+class WandbLogger(Logger):
+    r"""Log using `Weights and Biases <https://docs.wandb.ai/integrations/lightning>`_.
```

Comment:
I think this long docs shall have own page

Reply:
Hey @Borda! 
I am updating that linked page to show both the lightning and fabric workflows via selectable tabs. If I split the docs I will make sure to change the linked page!